android/src/main/kotlin/co/doneservices/callkeep/CallKeepNotificationManager.kt

```diff
@@ -0,0 +1,439 @@
+        }
+        notificationBuilder.setSound(missedCallSound)
+        notificationBuilder.setContentIntent(getAppPendingIntent(notificationId, data))
+        val accentColor = data.getString(EXTRA_CALLKEEP_ACCENT_COLOR, "#4CAF50")
+        try {
+            notificationBuilder.color = Color.parseColor(accentColor)
+        } catch (error: Exception) {
+        }
+
+        val notification = notificationBuilder.build()
+        getNotificationManager().notify(notificationId, notification)
+        Handler(Looper.getMainLooper()).postDelayed({
+            try {
+                getNotificationManager().notify(notificationId, notification)
+            } catch (error: Exception) {
+            }
+        }, 1000)
+    }
+
+
+    fun clearIncomingNotification(data: Bundle) {
+        context.sendBroadcast(IncomingCallActivity.getIntentEnded(context))
+        notificationId = data.getString(EXTRA_CALLKEEP_ID, "callkeep").hashCode()
+        getNotificationManager().cancel(notificationId)
+    }
+
+    fun clearMissCallNotification(data: Bundle) {
+        notificationId = data.getString(EXTRA_CALLKEEP_ID, "callkeep").hashCode()
+        getNotificationManager().cancel(notificationId)
+        Handler(Looper.getMainLooper()).postDelayed({
+            try {
+                getNotificationManager().cancel(notificationId)
+            } catch (error: Exception) {
+            }
+        }, 1000)
+    }
+
+    fun incomingChannelEnabled(): Boolean {
+        val notificationManager = getNotificationManager()
+        val channel = notificationManager.getNotificationChannel(NOTIFICATION_CHANNEL_ID_INCOMING)
+
+        return notificationManager.areNotificationsEnabled() &&
+                (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O &&
+                        channel != null &&
+                        channel.importance > NotificationManagerCompat.IMPORTANCE_NONE) ||
+                Build.VERSION.SDK_INT < Build.VERSION_CODES.O
+    }
+
+    private fun createNotificationChanel(
+        incomingCallChannelName: String,
+        missedCallChannelName: String,
+    ) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            var channelCall = getNotificationManager().getNotificationChannel(NOTIFICATION_CHANNEL_ID_INCOMING)
+            if (channelCall != null) {
+                channelCall.setSound(null, null)
+            } else {
+                channelCall = NotificationChannel(
+                        NOTIFICATION_CHANNEL_ID_INCOMING,
+                        incomingCallChannelName,
+                        NotificationManager.IMPORTANCE_HIGH
+                ).apply {
+                    description = ""
+                    vibrationPattern =
+                            longArrayOf(0, 1000, 500, 1000, 500)
+                    lightColor = Color.RED
+                    enableLights(true)
+                    enableVibration(true)
+                    setSound(null, null)
```

Comment:
Why no sound on incoming calls?

Reply:
As we play the sound from a background audio service, please check `CallKeepSoundPlayerService.kt`

Reply:
And this is mainly setting up the channels, so no need to set sound here, as we can do it on the notification builder (as we do for showing missed call notification)